src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+
+This is an interactive code notebook (a Jupyter Notebook). To run this code, click into each cell and press the ▶ button in the top toolbar, or press `shift+enter`.
+
+------------------------------------------------------------------------
+
+```{r}
+# Loading libraries
+suppressMessages({
+  library(ggplot2)
+  library(MGnifyR)
+  library(tidyverse)
+  library(vegan)
+  library(mia)
+  library(miaViz)
+  library(biomformat)
+  library(scater)
+  library(metagenomeSeq)
+  library(reshape2)
+  library(viridis)
+  library(patchwork)
+  library(IRdisplay)
+  library(ALDEx2)
+  library(ANCOMBC)
+  library(mikropml)
+  library(MLeval)
+  library(caret)
+})
+
+display_markdown(file = '../_resources/mgnifyr_help.md')
+```
+
+# Fetching data and preprocessing
+In this example we are going to fetch MGnify analysis results and metadata for a dataset on the human gut microbiome viewed across age and geography -  ([MGYS00005154](https://www.ebi.ac.uk/metagenomics/studies/MGYS00005154)).
+
+## Setting the study accession
+1. First, set the accession number of the study we're using to the `mgnify_study_accession` variable.
+
+```{r}
+#| output: false
+
+source("./lib/variable_utils.r")
+
+mgnify_study_accession <- get_variable_from_link_or_input('MGYS', 'Study Accession', 
+                                                          'MGYS00005154')
+
+# You can also just directly set the accession variable in code, like this:
+# mgnify_study_accession <- "MGYS00005292"
+```
+
+2. Then, construct a MgnifyClient object to access the database.
+
+```{r}
+#| output: false
+mg <- MgnifyClient(usecache = TRUE, cache_dir = '/home/jovyan/.mgnify_cache')
+```
+
+3. Fetch a list of the Analyses for the Study.
+
+```{r}
+#| output: false
+analyses_accessions <- searchAnalysis(mg, "studies", mgnify_study_accession)
+analyses_accessions
+```
+
+## Download metadata
+1. First, download the metadata for the first 20 Analyses and put it into a dataframe.
+
+```{r}
+#| output: false
+analyses_metadata_df <- getMetadata(mg, head(analyses_accessions,
+                                                              30))
+```
+
+2. The table could be big, so let's look at a sample of it (`head`)
+
+```{r}
+#| output: false
+t(head(analyses_metadata_df))
+```
+
+## Convert to [TreeSummarizedExperiment object using mia](https://microbiome.github.io/)
+
+```{r}
+#| output: false
+tse <- getResult(mg, analyses_metadata_df$analysis_accession)
+```
+
+# Normalization, alpha diversity indices and taxonomic profiles visualization
+## Cleaning the OTUs matrix
+1. First, identify the total counts per sample and add them to the colData of our `tse` object.
+
+```{r}
+tse <- addPerCellQC(tse)
+```
+
+2. We then plot a histogram to identify outliers, so we can remove samples with low coverage.
+
+```{r}
+library_sie_histogram <- ggplot(colData(tse)) +
```

Comment:
Does scater::plotColData work. (Let me know if not. It is rather common to plot histrograms so we could suggest the additin to scater package)

Reply:
In general also; say if something seems to be complex. These are rather common things to do, so usually there are simple solution available. (If not, then we can make one if a) the operation is common b) if the solution can be generalized)

Reply:
ok thanks will do, yes there will no doubt be methods that I've done too complicated - cause i was sorta looking at the phloseq examples, searching in the OMA book to find an OMA version of that thing... but there'll be points at which I've done something overly complex and missed a simpler method

Reply:
<img width="628" alt="Screenshot 2024-07-16 at 10 53 39" src="https://github.com/user-attachments/assets/eff37710-dcff-4c96-9f97-53588d8cd732">

The scatter plot with plotColData() creates something like this - what do you think? I think its less easy to find outliers than with a histogram, but still could be used